curator-framework/src/main/java/org/apache/curator/framework/imps/CuratorFrameworkImpl.java

```diff
@@ -637,7 +637,8 @@ boolean useContainerParentsIfAvailable()
         boolean doQueueOperation = false;
         do
         {
-            if ( RetryLoop.shouldRetry(event.getResultCode()) )
+            final KeeperException ke = KeeperException.create(event.getResultCode());
```

Comment:
nit - for me creating `ke` is unneeded. Create the exception directly as the argument